### DISCUSSION TITLE
Blogeintrag Woche 05 (Sem4) -  Testplan 2

### DISCUSSION THREAD

**Discussion by niklaskrauth** [mausio/corgi-shop-doc#15] [Blogs]
Im Backend wurden Unit-Tests, Datenbankentests und API-Tests hinzugefügt die verschiedene Abschnitte testet.

Weiterhin wurde ein Testplan für das Backend erstellt welche folgende Fragen beantwortet:

- Welche Arten von Tests werden eingesetzt, wie z. B. Unit-Tests, Integrationstests, API-Tests, Security-Tests usw.?
- Was ist der Zielwert für die Testabdeckung im Projekt?
- Wie werden die Testfälle verwaltet und ist es möglich, den Status der bestandenen/fehlgeschlagenen Tests basierend auf der Version des Quellcodes oder der Deployments nachzuverfolgen?

[Hier](https://github.com/mausio/corgi-shop-doc/blob/main/testing/backend.testing.md) sind Informationen zu dem Testen im Backend.

**Comment by ghost** [comment]
Sehr schön von euren Forschritten bei den Backend-Tests zu hören. Interessant wäre noch zu wissen, warum ihr euch dazu entschieden habt, eine Abdeckung von 80% anzustreben, anstelle einen anderen Zielwert festzulegen?

**Comment by JannisFuchs** [comment]
Super Plan. Allerdings frage ich mich warum ihr Security-Tests im Blogpost erwähnt wenn sie nicht eingeplant sind.

**Comment by lukashert** [comment]
Toll weiter so! 😲
